src/Components/Server/src/Circuits/CircuitHost.cs

```diff
@@ -127,11 +127,6 @@ public Task InitializeAsync(ProtectedPrerenderComponentApplicationStore store, C
                 // Now we wait for all components to finish rendering.
                 await Task.WhenAll(pendingRenders);
 
-                // At this point all components have successfully produced an initial render and we can clear the contents of the component
-                // application state store. This ensures the memory that was not used during the initial render of these components gets
-                // reclaimed since no-one else is holding on to it any longer.
-                store.ExistingState.Clear();
-
```

Comment:
This should stay

Reply:
Persisting state in blazor web doesn't work otherwise. The store is cleared too soon, before the components start rendering. Also the described example would work as before because when reading key from the state it gets erased https://github.com/dotnet/aspnetcore/blob/release/8.0/src/Components/Components/src/PersistentComponentState.cs#L121